src/main/java/utils/Storage.java

```diff
@@ -0,0 +1,92 @@
+package utils;
+
+import tasks.Deadline;
+import tasks.Event;
+import tasks.Task;
+import tasks.Todo;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * Represents a storage class that handles the reading and writing to the data text file
+ * for data persistence.
+ */
+public class Storage {
+    private File f;
+    private String filePath;
+    private DateTimeFormatter formatter = DateTimeFormatter.ofPattern("d/M/yyyy HHmm");
+
+    /**
+     * Class constructor with a given file path to the text file.
+     * @param filePath
+     */
+    public Storage(String filePath)  {
+        this.f = new File(filePath);
+        this.filePath = filePath;
```

Comment:
When the user first runs the program, the `/data` directory does not exist yet - we should create it in such a scenario to allow saving to work correctly.

```suggestion
        this.f = new File(filePath);
        f.getParentFile().mkdirs();
        this.filePath = filePath;
```